test/unit/esys-ac-send.c

```diff
@@ -0,0 +1,156 @@
+    TSS2_RC(*cancel) (TSS2_TCTI_CONTEXT * tctiContext);
+    TSS2_RC(*getPollHandles) (TSS2_TCTI_CONTEXT * tctiContext,
+                           TSS2_TCTI_POLL_HANDLE * handles,
+                           size_t * num_handles);
+    TSS2_RC(*setLocality) (TSS2_TCTI_CONTEXT * tctiContext, uint8_t locality);
+} TSS2_TCTI_CONTEXT_FAKE;
+
+static TSS2_RC
+tcti_fake_ac_send_transmit (
+    TSS2_TCTI_CONTEXT *tcti_ctx,
+    size_t size,
+    const uint8_t *cmd_buf)
+{
+    UNUSED(tcti_ctx);
+
+    TSS2_RC rc;
+    tpm_header_t header;
+    rc = header_unmarshal (cmd_buf, &header);
+    if (rc != TSS2_RC_SUCCESS) {
+        return rc;
+    }
+
+    assert_int_equal(header.code, TPM2_CC_AC_Send);
+
+    return TSS2_RC_SUCCESS;
+}
+
+static TSS2_RC
+tcti_fake_ac_send_receive (
+    TSS2_TCTI_CONTEXT * tctiContext,
+    size_t * response_size,
+    uint8_t * response_buffer, int32_t timeout)
+{
+    UNUSED(tctiContext);
+    UNUSED(response_size);
+    UNUSED(response_buffer);
+    UNUSED(timeout);
+
+    return TSS2_RC_SUCCESS;
+}
+
+static void
+tcti_fake_ac_send_finalize (TSS2_TCTI_CONTEXT * tctiContext)
+{
+    UNUSED(tctiContext);
+}
+
+static TSS2_RC
+tcti_fake_initialize(TSS2_TCTI_CONTEXT * tctiContext, size_t * contextSize)
+{
+    TSS2_TCTI_CONTEXT_FAKE *tcti_fake =
+        (TSS2_TCTI_CONTEXT_FAKE *) tctiContext;
+
+    if (tctiContext == NULL && contextSize == NULL) {
+        return TSS2_TCTI_RC_BAD_VALUE;
+    } else if (tctiContext == NULL) {
+        *contextSize = sizeof(*tcti_fake);
+        return TSS2_RC_SUCCESS;
+    }
+
+    /* Init TCTI context */
+    memset(tcti_fake, 0, sizeof(*tcti_fake));
+    TSS2_TCTI_MAGIC(tctiContext) = TCTI_FAKE_MAGIC;
+    TSS2_TCTI_VERSION(tctiContext) = TCTI_FAKE_VERSION;
+    TSS2_TCTI_TRANSMIT(tctiContext) = tcti_fake_ac_send_transmit;
+    TSS2_TCTI_RECEIVE(tctiContext) = tcti_fake_ac_send_receive;
+    TSS2_TCTI_FINALIZE(tctiContext) = tcti_fake_ac_send_finalize;
+    TSS2_TCTI_CANCEL(tctiContext) = NULL;
+    TSS2_TCTI_GET_POLL_HANDLES(tctiContext) = NULL;
+    TSS2_TCTI_SET_LOCALITY(tctiContext) = NULL;
+
+    return TSS2_RC_SUCCESS;
+}
+
+static int
+setup(void **state)
+{
+    TSS2_RC r;
+    ESYS_CONTEXT *ectx;
+    size_t size = sizeof(TSS2_TCTI_CONTEXT_FAKE);
+    TSS2_TCTI_CONTEXT *tcti = malloc(size);
+
+    r = tcti_fake_initialize(tcti, &size);
+    if (r)
+        return (int)r;
+    r = Esys_Initialize(&ectx, tcti, NULL);
+    *state = (void *)ectx;
+    return (int)r;
+}
+
+static int
+teardown(void **state)
+{
+    TSS2_TCTI_CONTEXT *tcti;
+    ESYS_CONTEXT *ectx = (ESYS_CONTEXT *) * state;
+    Esys_GetTcti(ectx, &tcti);
+    Esys_Finalize(&ectx);
+    free(tcti);
+    return 0;
+}
+
+static void
+test_ac_send(void **state)
+{
+    //TSS2_RC r;
+    ESYS_CONTEXT *ectx = (ESYS_CONTEXT *) * state;
+
+    Esys_AC_Send(ectx, 0, 0, ESYS_TR_NONE,
+                     ESYS_TR_NONE, ESYS_TR_NONE, 0, NULL, NULL);
+
+    //assert_int_equal(r, TSS2_RC_SUCCESS);
```

Comment:
why no checking of return code? You can just have send/receive function build a successful return buffer and copy it in. I ran a command that has nothing but a response code and grabbed the return with pcap TCTI.
```bash
tpm2_dictionarylockout --tcti=pcap -c
```
However the response from tpm2_flushcontext is better (which is run within the tpm2_dictionarylockout tool) becuase it doesn't have a session.

```hex
80 01 00 00 00 0a 00 00 00 00
```

The response is:
- 0x8001 (The tag no sessions)
- 0x0000000a (The size of the whole thing 10 bytes)
- 0x00000000 (The RC of 0 or TPM2_RC_SUCCESS)